test/ipld-all.js

```diff
@@ -31,92 +29,61 @@ describe('IPLD Resolver for dag-cbor + dag-pb', () => {
   let cidCbor
   let cidPb
 
-  before((done) => {
-    waterfall([
-      (cb) => inMemory(IPLDResolver, cb),
-      (res, cb) => {
-        resolver = res
-        dagPB.DAGNode.create(Buffer.from('I am inside a Protobuf'), cb)
-      },
-      (node, cb) => {
-        nodePb = node
-        dagPB.util.cid(nodePb, cb)
-      },
-      (cid, cb) => {
-        cidPb = cid
-        nodeCbor = {
-          someData: 'I am inside a Cbor object',
-          pb: cidPb
-        }
-
-        dagCBOR.util.cid(nodeCbor, cb)
-      },
-      (cid, cb) => {
-        cidCbor = cid
-
-        each([
-          { node: nodePb, format: multicodec.DAG_PB, cidVersion: 0 },
-          { node: nodeCbor, format: multicodec.DAG_CBOR, cidVersion: 1 }
-        ], (nac, cb) => {
-          resolver.put(nac.node, nac.format, {
-            cidVersion: nac.cidVersion
-          }).then(
-            () => cb(null),
-            (error) => cb(error)
-          )
-        }, cb)
-      }
-    ], done)
+  before(async () => {
+    resolver = await new Promise((resolve, reject) => {
+      inMemory(IPLDResolver, (error, res) => {
```

Comment:
this looks easily promisifiable, up to you

Reply:
Sadly isn't promisifiable. According to Alan because it thinks IPLDResolver is function, hence a callback.